### PR TITLE
fix incremental decoder bug

### PR DESCRIPTION
**Pull request by naxingyu**
@chenzhehuai 
not sure if this is what you meant?

### CODE REVIEW

**Comment by naxingyu** [comment]
oh, I should @danpovey since you wrote this.

File: src/decoder/lattice-incremental-online-decoder.cc [hunk 1]
```diff
@@ -98,7 +98,7 @@ typename LatticeIncrementalOnlineDecoderTpl<FST>::BestPathIterator LatticeIncrem
     // it a fatal error for now.
     KALDI_WARN << "No final token found.";
   }
-  if (final_cost_out == NULL)
+  if (final_cost_out != NULL)
     *final_cost_out = best_final_cost;
   return BestPathIterator(best_tok, this->NumFramesDecoded() - 1);
 }
```
Comment (danpovey) [inline src/decoder/lattice-incremental-online-decoder.cc]: yes but please do the same for lattice-incremental-decoder.cc

Comment (naxingyu) [reply src/decoder/lattice-incremental-online-decoder.cc]: I do not see this error in lattice-incremental-decoder.cc, where exactly?

Comment (danpovey) [reply src/decoder/lattice-incremental-online-decoder.cc]: oh sorry, it's just here.

**Pull request merged**